practice-two/src/services/customerService.js

```diff
@@ -1,12 +1,16 @@
-import { BASE_URL, PATH } from "@constants";
-
-const getAllCustomerService = async () => {
-  try {
-    const response = await fetch(`${BASE_URL}/${PATH}`);
-    return response.json();
-  } catch (error) {
-    return new Error("Sorry! Having some error, please try again");
+// import { BASE_URL, PATH } from "@constants";
+
+const getAllCustomerService = async (url) => {
+  const response = await fetch(url);
+
+  if (!response.ok) {
+    const error = new Error("An error occurred while fetching the data.");
+    error.info = await response.json();
+    error.status = response.status;
```

Comment:
why need to throw info and status error?

Reply:
If I assign info and status to the error, I can handle the error more explicitly and provide more detailed information about why the HTTP request failed.

Reply:
when you throw this error message for end user, the user don't know what is the error message. It'll make us confuse.
Just need to return data or error messages

Reply:
I changed it just throw message when occuring error
![image](https://github.com/datdragon777/React-Training/assets/74806826/bdd29220-c77c-405e-b74b-12bb8ee9d83b)